src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -4514,8 +4514,11 @@ private MemberAst ClassMemberRule(string className, out List<Ast> astsOnError)
                 }
             }
 
-            if (token.Kind == TokenKind.Identifier)
+            if (token.Kind == TokenKind.Identifier
+                || token.TokenFlags.HasFlag(TokenFlags.Keyword))
             {
+                token.TokenFlags = TokenFlags.None;
```

Comment:
It would be great if we can use [`SimpleNameRule`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/parser/Parser.cs#L2131) here to get the method name, just like how we parse the class name, so the token we got back is accurate. See the two `class` tokens below for an example.
```
$tokens = $null
$s = [System.Management.Automation.Language.Parser]::ParseInput("class class {  }", [ref]$tokens, [ref]$null)
PS:20> $tokens

Text       : class
TokenFlags : Keyword
Kind       : Class
HasError   : False
Extent     : class

Text       : class
TokenFlags : TypeName
Kind       : Identifier
HasError   : False
Extent     : class

...
```
See how `ClassDefinitionRule` does it:
https://github.com/PowerShell/PowerShell/blob/981c990d747ae97f56a6c7474e5d2ef04bf237c5/src/System.Management.Automation/engine/parser/Parser.cs#L4156-L4169

However, that would require we resync the token because the token has already been read before setting `WantSimpleName`, and `Resync` is relatively expensive.
I guess I'm fine we just check for `Keyword` and `DynamicKeyword` here as a special case of `SimpleNameRule`, but I'm not sure if `SimpleNameRule` covers some other token cases as well.

Note that, you handled `keyword` but not the `DynamicKeyword`, see the code in `Tokenizer`:
https://github.com/PowerShell/PowerShell/blob/981c990d747ae97f56a6c7474e5d2ef04bf237c5/src/System.Management.Automation/engine/parser/tokenizer.cs#L4371-L4385

Reply:
Heh, I was poking at that, seeing if there was a way to make a tokenizer exception for keywords in the specific context of a method name, but I couldn't find anything that seemed to work.

But yeah, we can definitely add `DynamicKeyword` as well. Good point!

Reply:
The way `SimpleNameRule` works is to set the special `WantSimpleName` flag and then go on with the next tokenizing work, which requires that no token as been peeked yet.
In our case here, we already peeked the token, which means the token was read with `WantSimpleName = false`. To use `SimpleNameRule`, we will have to resync back to the start offset of the peeked token.

Reply:
Hmm, I see.

Let me examine this a bit more closely. There's probably no better way, but I want to make sure. 🙂

Reply:
In case there is no better way (meaning we have to resync in order to leverage `WantSimpleName` flag in the tokenizer), then maybe it makes sense to make it a private utility method to alter a token that is to be used as a simple name, so the code can be reused in other rules.

Reply:
That was my backup thought as well! 🙂

Reply:
Yep, looks like that'll have to be the way forward, short of recoding that whole section of the tokenizer _and_ parser. Probably only to arrive at a similar issue somewhere else anyway. 😄

Reply:
As for `DynamicKeyword`  - Could this complicate the readability of scripts too much?
For keywords PSSA should caution I think.